infra/src/main/kotlin/fi/fta/geoviite/infra/configuration/RequestFilter.kt

```diff
@@ -257,6 +300,27 @@ class RequestFilter @Autowired constructor(
             ?: throw IllegalArgumentException("Invalid key (expected ECPublicKey): ${generated::class.qualifiedName}")
         return Algorithm.ECDSA256(publicKey, null)
     }
+
+    private fun isIntegrationApiRequest(request: HttpServletRequest): Boolean {
+        return environment.activeProfiles.contains("integration-api") &&
+            request.requestURI.startsWith("/rata-vkm")
+    }
```

Comment:
API-käyttäjää yritetään määrittää vain kun tuo profiili on käytössä, sekä lisärajauksena laitettu toistaiseksi tuo rata-vkm:n juuripolku.

Teoriassa jos tuota juuripolkua rata-vkm -rajapintaan ei olisi ja springi startattaisiin backend- JA integration-api profiililla (ei-lokaalisti), niin kävisi hassusti: kaikki käyttäjienkien pyynnöt normirajapintoihinkin todettaisiin integration-api pyynnöiksi. Privileget toki rajaavat ettei aiempiin backend-rajapintoihin silloin tietysti pääsisi kuitenkaan käsiksi (tämä oli se asia jota Juuse mietti slackin puolella)

Tähän saattaa myös olla jokin mielekkäämpikin tapa, mutta vaikutti toistaiseksi suht yksinkertaiselta tehdä tämä näin.

Reply:
Hyvä että rajattu. Mietin tosin kannattaisiko tässä esimerkiksi luoda erilliset tarkastelut skipAuthille ja authatulle tilanteelle. Lokaalia bäkkäriä skipAuthilla ajettaessa mielestäni mikä tahansa `/rata-vkm`-alkuiseen osoitteeseen lähetetty kutsu saisi lukeutua vkm-kutsuksi (testaaminen pysyy helppona kun vkm:n ja ei-vkm:n voi testata samalla lokaalilla bäkkärillä, mutta jos/kun vkm-toiminnallisuden päälläoloa kontrolloidaan profiililla, niin se mahdollistaa myös niiden tilanteiden testaamisen missä bäkkäri on päällä mutta vkm ei.) Vastaavasti AWS:stä ajettuna mikä tahansa API-kontille tuleva pyyntö saisi mielestäni lukeutua viitekehysmuunninkutsuksi kolkutellusta rajapinnasta riippumatta (API-kontille ei pitäisi päätyä mitään normikonttien kutsuja, ja tämä olisi yksi lisävarmistus sille)

Reply:
Tämä on ehkä vähän spesiaali case, mutta yleensä aina kun jossain tarkastellaan että "jos profiili on x" kannattaisi sen sijaan olla property "y.enabled" tms ja asettaa se kyseisen profiilin sisällä. Silloin konfiguraatio on profiilin konffissa kaikki samassa muodossa eikä pitkin koodia iffeinä. Tämäkin voisi ihan hyvin olla geoviite.integration-api.enabled

Reply:
Jeppis, tuo .enabled vaikuttaakin varsin loogiselta nyt kun mainitsit, edellyttää tosin muutoksia muutamaan paikkaan joten muuttelen eri pullarissa.

Ja jos Juuse hait sitä että lokaalisti voi ajaa samaan aikaan normibackendiä ja integration-apia, niin tämä auth-rakenne onkin toteutettu niin että se jo toimiikin oletetulla tavalla. Tuo automaattimappäys vkm-pyynnöksi AWS:n puolella on toki mahdollista tehdä, mutta jotenkin tällainen white-/allow-listaus tuntuu omaan mieleen turvallisemmalta, vaikkei käytännössä eroa olisikaan ja turvallisuuden määrä ei muutu 🤔. Tulkinta ehkä kuitenkin hieman helpottuu eksplisiittisyyden vuoksi.

Reply:
Meinasin todennäköisesti hieman eri juttua (meinasin että lokaalisti ajettuna sama bäkkäri voisi vastata molempiin, ja sitten se vaan valkkaisi puhtaasti urlin perusteella minkälaisena käyttäjänä näytään.) Mutta tuo Jyrkin ehdottama `enabled`-propsu on mielestäni enivei ehdotetuista vaihtoehdoista paras.

Reply:
Aivan, tosin hyvin lähellehän se noin toimii jo nytkin. Tässä Requestfiltterissähän tuolla ylempänä tutkaillaan skipAuth-profiilia lokaalin käyttäjän määritystä, mikä yliajaa kaikki myöhemmät if-chekit. Toistaiseksi siis tätä funkkaria vaan kutsutaan kahdesta paikasta (non-local ja local-checkit ovat ns. erillisiä).

Lyhyesti siis jos skipAuth on päällä -> kutsutaan tätä funkkaria -> jos true, niin lokaali käyttäjä on integration-api käyttäjä ja muuten normibackendin käyttäjä